File: backend/Services/Parsers/MinuteCrypticParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class MinuteCrypticParser : BasicScoreResultParser
    {
        private ILogger<MinuteCrypticParser> _logger;

        public MinuteCrypticParser(ILogger<MinuteCrypticParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Minute Cryptic";
        public override string? HelpText => null;
        // ⚪️
        // 🟣 = \uD83D\uDFE3
        // 🟡 = \uD83D\uDFE1
        protected override Regex Parser => new Regex($"{GameName}.+?(?<{ScoreGroup}>(⚪️|🟡|🟣)+)", RegexOptions.Singleline);
        protected override string? ExtraContent => Url + "/?utm_source=share";
        public override string Url => "https://www.minutecryptic.com";
        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (!parserResults.Groups.ContainsKey(ScoreGroup))
            {
                return dailyResult;
            }

            var successIndex = parserResults.Groups[ScoreGroup].Value.IndexOf("🟣");
            if (successIndex > -1)
            {
                dailyResult.Score = successIndex / 2;
            }

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/minutecryptic-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class MinuteCrypticParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Minute Cryptic';
  readonly helpText = null;
  protected readonly extraContent = this.url + '/?utm_source=share';
  readonly url = 'https://www.minutecryptic.com';
  
  // Match emoji circles: ⚪️, 🟡, 🟣
  protected readonly parser = new RegExp(
    `${this.gameName}.+?(?<${BasicScoreResultParser.SCORE_GROUP}>(⚪️|🟡|🟣)+)`,
    's'
  );
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    if (!hasGroup(match, BasicScoreResultParser.SCORE_GROUP)) {
      return dailyResult;
    }
    
    const emojiStr = getGroup(match, BasicScoreResultParser.SCORE_GROUP);
    if (emojiStr) {
      // Score is the position of the purple circle (🟣) divided by 2
      // Each emoji is 2 characters in the string
      const successIndex = emojiStr.indexOf('🟣');
      if (successIndex > -1) {
        dailyResult.score = successIndex / 2;
      }
    }
    
    return dailyResult;
  }
}
